app/controllers/matches_controller.rb

```diff
@@ -0,0 +1,14 @@
+class MatchesController < ApplicationController
+
+  def show
+    @my_item = Item.find_by(id: params[:item_id])
+    @match = Match.includes(:messages).find(params[:id])
+    if @my_item && @my_item.user == current_user && @match.item_belongs_to?(@my_item)
+      @other_item = @match.other(@my_item)
+      @message = Message.new
+    else
+      render :file => "#{Rails.root}/public/404.html", :status => 404
```

Comment:
Again it might be preferable here to raise ActiveRecordNotFound and allow rails to do its default behavior (which in production mode is to return a 404 and in development mode is to throw up the error page.)